pre-docs/inventory_version2.md

```diff
@@ -76,6 +76,93 @@ plan setup_lb (
 }
 ```
 
+## Creating targets in plans
+
+When using inventory v2 a new and improved API for interacting with `Target`s in inventory is available. Two new plan functions have been added: `get_target` which allows retrieving a single `Target` from inventory and `set_config` which allows setting config on a specific `Target`. The updated API also provides a way to instantiate new `Target`s with data that more closely resembles how targets are declared in an inventory file.
+
+For example, consider the following `new_targets` plan:
+```
+plan new_targets(){
+  $new_target = get_target('ssh://user:secret@1.2.3.4:2222')
+  $new_target.set_config(['ssh', 'host-key-check'], false)
+}
+```
+In the `new_targets` plan the `get_target` function returns a `Target` identified with the name `ssh://user:secret@1.2.3.4:2222`. If a `Target` with that name does not exist in inventory a new `Target` is instantiated with the `uri` and `name` attributes set to `ssh://user:secret@1.2.3.4:2222` and is added to the `all` group in inventory (where it inherits and configuration for the `all` group. If the `Target` with that name does exist, then it is simply returned. The `set_config` method is used to set a transport specific setting specified by the array of keys to that setting that matches the keys in the structured hash found in an inventoryfile under the `config` key. This illustrates how a new `Target` can be created from a URI and configuration options that are not able to be set in URI parts can be modified. 
+
+The `Target.new` method may also be used to instantiate a `Target`:
+```
+plan new_target_alternate(){
+  $config = { 'transport' => 'ssh',
+              'ssh' => {
+                'user' => 'user',
+                'password' => 'secret',
+                'host' => '1.2.3.4'
+                'port' => 2222,
+                'host-key-check' => false
+                }}
+  $new_target = Target.new('name' => 'new_target', 'config' => $config)
```

Comment:
2 things: 1. I'd add a line explaining what this plan does (just like "This plan creates a new target using `Target.new` and passing in a config hash") 2. I'd add another example that also passes in features or vars or facts